src/vec_cast.h

```diff
@@ -85,16 +82,24 @@ inline void pack(
     assert(sizeof(Td) >= word_size);
     assert(word_size % sizeof(Ts) == 0);
     // get only word_size bytes from each element
-    if (word_size == 1) {
+    switch (word_size) {
```

Comment:
Out of the scope of this PR, but I wonder if `word_size` could be a template parameter.

In other word, does it make sense to change word_size after instanciating the FEC object?